### PR TITLE
Add disk driver support for RT600

### PR DESCRIPTION
**Pull request by mmahadevan108**
Tested using `samples/subsys/fs/fat_fs`

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @nxf58843

File: drivers/disk/usdhc.c [hunk 1]
```diff
@@ -2634,35 +2634,42 @@ static int usdhc_board_access_init(struct usdhc_priv *priv)
...
 	if (!priv->detect_gpio) {
-		LOG_INF("USDHC detection other than GPIO not implemented!");
-		return 0;
-	}
+		LOG_INF("USDHC detection other than GPIO");
+		/* DATA3 does not monitor card insertion */
+		base->PROT_CTRL &= ~USDHC_PROT_CTRL_D3CD_MASK;
+		if ((base->PRES_STATE & USDHC_PRES_STATE_CINST_MASK) != 0) {
+			priv->inserted = true;
+			ret = 0;
```
Comment (dleach02) [inline drivers/disk/usdhc.c:2642]: just curious but why is this needed if you are going to change the declare of ret to be initialized to 0?

Comment (mmahadevan108) [reply drivers/disk/usdhc.c]: Thanks for catching this, I have removed this line.

File: drivers/gpio/gpio_mcux_lpc.c [hunk 2]
```diff
@@ -374,6 +374,9 @@ static const clock_ip_name_t gpio_clock_names[] = GPIO_CLOCKS;
 #define GPIO_MCUX_LPC_IRQ(n, m)								\
 	COND_CODE_1(DT_INST_IRQ_HAS_IDX(n, m), (GPIO_MCUX_LPC_IRQ_CONNECT(n, m)), ())
 
+#define GPIO_MCUX_LPC_IRQ_DATA(n)								\
+	COND_CODE_1(DT_INST_IRQ_HAS_IDX(n, 0), (struct gpio_mcux_lpc_data *data = dev->data), ())
```
Comment (MaureenHelm) [inline drivers/gpio/gpio_mcux_lpc.c:378]: I think it would be clearer to move this into `GPIO_MCUX_LPC_IRQ_CONNECT`

File: drivers/disk/usdhc.c [hunk 3]
```diff
@@ -2634,35 +2634,41 @@ static int usdhc_board_access_init(struct usdhc_priv *priv)
 	}
 
 	if (!priv->detect_gpio) {
-		LOG_INF("USDHC detection other than GPIO not implemented!");
-		return 0;
-	}
+		LOG_INF("USDHC detection other than GPIO");
```
Comment (MaureenHelm) [inline drivers/disk/usdhc.c:2637]: Please update the commit message `drivers: disk: Update NXP USDHC driver` to indicate that this update is to support the IP version on RT600

File: soc/arm/nxp_imx/rt6xx/soc.h [hunk 4]
```diff
@@ -75,4 +75,21 @@
 /*!<@brief Slow mode */
 #define IOPCTL_PIO_SLEW_RATE_SLOW 0x80u
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if CONFIG_DISK_DRIVER_SDMMC &&					\
+	(DT_NODE_HAS_STATUS(DT_NODELABEL(usdhc1), okay) ||	\
+	 DT_NODE_HAS_STATUS(DT_NODELABEL(usdhc1), okay))
```
Comment (MaureenHelm) [inline soc/arm/nxp_imx/rt6xx/soc.h:84]: This should be `usdhc2`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/disk/usdhc.c (hunk 3 above)
Comment (mmahadevan108) [reply drivers/disk/usdhc.c:2637]: This is available to USDHC IP version present on RT1064 and other RT platform. This pin was not used in those SoC's, instead the detect signal was connected to a GPIO pin.

File: soc/arm/nxp_imx/rt6xx/soc.h (hunk 4 above)
Comment (mmahadevan108) [reply soc/arm/nxp_imx/rt6xx/soc.h:84]: Thank you, I have fixed this. I noticed the same error exists in the rt implementation. I have added a separate commit to fix this as well.

File: drivers/gpio/gpio_mcux_lpc.c (hunk 2 above)
Comment (mmahadevan108) [reply drivers/gpio/gpio_mcux_lpc.c]: Done

**Review by MaureenHelm — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]
